### ISSUE TITLE
Accessibility Calendar. Click arrow to select day

### ISSUE THREAD

**Issue by denant0** [grommet/grommet#5677]
Using Calendar, Low contrast grey button against the white background is still the only visual means of tracking keyboard focus inside the calendar.  What can be done so that when you press the keys, the text is read?

### Expected Behavior

![изображение](https://user-images.githubusercontent.com/1597545/136600618-1631c46b-6b7d-4b4c-8cbf-d08d8d50a239.png)

**Comment by jcfilben** [comment]
When using a screenreader the dates are correctly read out when navigating, this was fixed in https://github.com/grommet/grommet/pull/5319. 

I agree that the grey button against the white background isn't enough of a visual indicator for tracking keyboard focus. I think we should have the focus indicator on the date when navigating with the keyboard rather than having the focus indicator on the whole calendar.

**Comment by Ojas-Sri** [comment]
Hi,

I have made some progress in addressing this issue. Below are 2 screenshots for the same , please suggest if any changes are needed:

(The calendar will be in focus when mouse clicks on the calendar area. When the keyboard navigates to other days, the focus moves from calendar to the particular day. If the user clicks again via mouse pointer , focus resets to calendar)

![PR (11)](https://user-images.githubusercontent.com/80472243/161683492-f7c76be9-963a-423d-a03d-d1b2077a9ff2.gif)

![PR (12)](https://user-images.githubusercontent.com/80472243/161683613-cd970580-366c-447d-8788-8c7b83eb1d47.gif)

**Comment by jcfilben** [comment]
Hi @Ojas-Sri nice work on this and thanks for sharing what you are thinking
This behavior looks good to me. I just had a few questions
- If a user is navigating by keyboard and tabs to the calendar will the focusIndicator be around the whole calendar or on the first day of the month? I'm thinking it should be on the first day of the month in this case
- When the focusIndicator is on a specific date it looks like the calendar still has an outline around it. I think if we move the focus indicator to a date the calendar shouldn't have an outline

**Comment by Ojas-Sri** [comment]
Hi @jcfilben,

Thanks for the comments, I have addressed both of them. Screenshot of my implementation attached below.

![PR (2)](https://user-images.githubusercontent.com/80472243/162043958-a5b0c317-5adf-4d2f-a201-7999ac816f07.gif)


![PR (3)](https://user-images.githubusercontent.com/80472243/162044138-8cedee4a-1b73-479f-8c58-3382d39dc13a.gif)


![PR (3)](https://user-images.githubusercontent.com/80472243/162044327-d90c407e-ec4c-4fd5-bac4-fd482ee310b0.gif)

**Comment by jcfilben** [comment]
@Ojas-Sri this looks good to me! Feel free to file a pull request for this

**Comment by jcfilben** [comment]
We need to discuss https://github.com/grommet/hpe-design-system/issues/2664 before working on this issue
